source/common/http/filter_manager.cc

```diff
@@ -908,6 +908,13 @@ void DownstreamFilterManager::sendLocalReply(
       !filter_manager_callbacks_.informationalHeaders().has_value()) {
     // If the response has not started at all, send the response through the filter chain.
 
+    if (auto cb = filter_manager_callbacks_.downstreamCallbacks(); cb.has_value()) {
+      // The initial route maybe never be set or the cached route maybe cleared by the filters.
+      // This will force route refreshment if there is not a cached route to avoid potential
+      // route refreshment in the response filter chain.
+      cb->route(nullptr);
```

Comment:
So this is a NOP if the route is already cached? Just making sure we aren't doing extra work on this path for the common case?

Reply:
Yeah, if a route is already cached, then the route will return directly and this calling will do nothing.